devicetypes/smartthings/ecobee-thermostat.src/ecobee-thermostat.groovy

```diff
@@ -185,6 +185,11 @@ def generateEvent(Map results) {
 				isChange = isStateChange(device, name, value.toString())
 				event['isStateChange'] = isChange
 				event['displayed'] = false
+			} else if (name == "thermostatMode") {
+				def mode = value.toString()
+				mode = (mode == "auxHeatOnly") ? "emergency heat" : mode
```

Comment:
this would be better as an if-else. logic is right but makes code reading difficult